server/aws-lsp-codewhisperer/src/language-server/utilities/filesystemUtilities.ts

```diff
@@ -0,0 +1,34 @@
+// Port of implementation in AWS Toolkit for VSCode
+// https://github.com/aws/aws-toolkit-vscode/blob/ac9a78db90192fe13dcd82ac0b30e6f84c39b9c8/packages/core/src/shared/filesystemUtilities.ts
+
+import * as pathutils from './pathUtils'
+
+/**
+ *
+ * @returns file distance between fileA and fileB
+ * For example:
+ * The file distance between A/B/C.java and A/B/D.java is 0
+ * The file distance between A/B/C.java and A/D.java is 1
+ */
+export function getFileDistance(fileA: string, fileB: string): number {
+    // TODO: Test how it works with nested folders with spaces (%20 symbol)
```

Comment:
This todo is not coming from the toolkit codebase but something we are introducing, why is that?

Reply:
this is something I noticed when comparing logs for VSCode Q and Language Server. In Language server, files URIs are encoded and spaces are converted to `%20` string. Since this is different from original environment, I wanted to double check that in this case this function works. I added few tests to cover this and will remove this comment,